www/ec2/ec2.inc.php

```diff
@@ -158,9 +166,16 @@ function EC2_TerminateIdleInstances() {
       }
     }
 
+	$idleTerminateMinutes = GetSetting("EC2.IdleTerminateMinutes");
+	$minutes = $instance['runningTime'] / 60.0;
+	if ($idleTerminateMinutes) {
+		$timeCheck = ( $minutes > $idleTerminateMinutes );

```

Comment:
This just checks that it has been running for longer than IdleTerminateMinutes, not that it has been idle that long.  The actual idle check is below where $terminate=false is set.  It is still hard-coded to not terminate if it got work in the last 15 minutes.

If that's intended then the setting is probably better to be called "EC2.CheckTerminateMinutes"

Reply:
Ah, I see what you mean.  I missed that clause further down.  I missed this because I was doing a tight test with a single instance, so the real idle check wasn't always being applied.

So something like this:

```
diff --git a/www/ec2/ec2.inc.php b/www/ec2/ec2.inc.php
index 6ac24f77..69a206a8 100644
--- a/www/ec2/ec2.inc.php
+++ b/www/ec2/ec2.inc.php
@@ -194,9 +194,9 @@ function EC2_TerminateIdleInstances() {
           }
         }
         
-        // Keep the instance if the location had work in the last 15 minutes
-        // and if this instance has checked in recently
-        if (isset($lastWork) && isset($lastCheck) && $lastWork < 15 && $lastCheck < 15)
+        // Keep the instance if the location had work in the last^M
+        // EC2.IdleTerminateMinutes and if this instance has checked in recently^M
+        if (isset($lastWork) && isset($lastCheck) && $lastWork < $idleTerminateMinutes && $lastCheck < $idleTerminateMinutes)^M
           $terminate = false;
         
         if ($terminate) {
```

Does that look reasonable?

Reply:
That should work.  Might still not be a bad idea to have a separate setting for the minimum instance run time.

Reply:
Sorry, one more issue.  The check needs to be inside of the foreach loop because $minutes is derived from the instance data.  The legacy path is changed otherwise (broken).